inc/admin/metabox/src/components/controls/CheckBox.js

```diff
@@ -0,0 +1,39 @@
+import {compose} from "@wordpress/compose";
+import {withDispatch, withSelect} from "@wordpress/data";
+import {CheckboxControl} from "@wordpress/components";
+
+export let CheckBox = compose(
+	withDispatch((dispatch, props) => {
+		return {
+			setMetaFieldValue: (value, dependent) => {
+
+				if ( dependent ){
+					let control = document.getElementById(dependent);
+					if( value === true ){
+						control.style.display = "block";
+					} else {
+						control.style.display = "none";
+					}
+				}
+				dispatch('core/editor').editPost( { meta: { [props.id]: ( value ? 'on' : 'off' ) } } );
+			}
+		};
+	}),
+	withSelect((select, props) => {
+		return {
+			metaFieldValue: select('core/editor').getEditedPostAttribute('meta')[props.id]
+		};
+	})
+)((props) => {
```

Comment:
props can be destructured inside the arrow function:

Instead of `(props)` you can simply write it as `({data, metaFieldValue, setMetaFieldValue})` as it is much easier to know at first sight what props the component receives. 

After that you can simply use them instead of `props.data`, `props.metaFieldValue`, etc.